test/getArgumentsTest.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+import test from 'ava'
+
+const getArguments = require('../lib/getArguments')
+let route = '/user/:id'
+
+test('it doesn\'t spit out a name with a slash', t => {
+  let placeholder = 'iamthe/last'
+  let url = '/user/' + placeholder
+  t.notSame(getArguments(url, route), ['iamthe, last'])
+})
+
+test('it produces the correct string/integer argument', t => {
+  let placeholder = 'iamthe1st'
+  let url = '/user/' + placeholder
+  t.same(getArguments(url, route), ['iamthe1st'])
+})
+
+test(`it spits out a name with a full stop`, t => {
+  let placeholder = 'iamthe.1st'
+  let url = '/user/' + placeholder
+  t.same(getArguments(url, route), ['iamthe.1st'])
+})
+
+test(`it spits out a name with an underscore`, t => {
+  let placeholder = 'iamthe1_st'
+  let url = '/user/' + placeholder
+  t.same(getArguments(url, route), ['iamthe1_st'])
+})
+
+test('it returns placeholders from random strings', t => {
+  const randomUrlsWithPlaceholders = require('../lib/randomUrlsWithPlaceholders.js')
+  const { url, route } = randomUrlsWithPlaceholders()
+  let i = 0
+// split string to find placeholder, count number and count number of args returned
+  let routeWords = route.split('/')
+  routeWords.forEach((section, index) => {
```

Comment:
You can use [String.prototype.match](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match) to get a list of matches for a regex, rather than using a foreach & counting them manually